lib/value-pairs/transforms.c

```diff
@@ -152,6 +151,39 @@ value_pairs_new_transform_shift (gint amount)
   return (ValuePairsTransform *)vpt;
 }
 
+/* shift-levels() */
+
+static void
+vp_trans_shift_levels(ValuePairsTransform *t, GString *key)
+{
+  VPTransShift *self = (VPTransShift *)t;
+  const gchar *dot;
+  gint levels_left = self->amount - 1;
+
+  dot = strchr(key->str, '.');
+  while (dot && levels_left > 0)
+    {
+      dot = strchr(dot + 1, '.');
+      levels_left--;
+    }
+
+  g_string_erase(key, 0, dot + 1 - key->str);
```

Comment:
Is there always dot in a name? If not, this will assert in glib.

Reply:
fixed. dot is not guaranteed.